packages/babel-parser/src/parser/expression.js

```diff
@@ -2234,9 +2235,18 @@ export default class ExpressionParser extends LValParser {
       );
     }
 
+    if (!this.scope.inFunction && !this.options.allowAwaitOutsideFunction) {
+      if (this.hasPrecedingLineBreak()) {
```

Comment:
I think the following examples are also unambiguous
```js
await + 0
await - 0
await ( 0 )
await [ 0 ]
await / 0 /u
```

Reply:
I will try to fix it without using lookahead if possible, to avoid special casing different char codes (which can cause errors if we forget some of them / add new ones)